### PR TITLE
Insert a space before colon in ternary expression in Fix-It

### PR DESCRIPTION
**Pull request by gibachan**
This pull request addresses #1607

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — COMMENTED** [review]
@gibachan, welcome to SwiftSyntax’s development. 

I would prefer to adjust the formatting in `BasicFormat` instead of `ParseDiagnsoticsGenerator` because that way we also get the correct formatting you you are constructing a ternary operator e.g. from `SwiftSyntaxBuilder`, which is what people will do when writing macros. 

In that case, it should work both when the operators are folded and when they haven’t been folded yet (see https://swiftpackageindex.com/apple/swift-syntax/main/documentation/swiftoperators).

An unfolded ternary operator would be the following

```swift
SequenceExprSyntax {
  BooleanLiteralExprSyntax(true)
  UnresolvedTernaryExprSyntax(firstChoice: IntegerLiteralExprSyntax(1))
  IntegerLiteralExprSyntax(0)
}
```

and a folded ternary operator would be

```swift
TernaryExprSyntax(
  conditionExpression: BooleanLiteralExprSyntax(true),
  firstChoice: IntegerLiteralExprSyntax(1),
  secondChoice: IntegerLiteralExprSyntax(0)
)
```

The code that you’d like to change is most likely in here https://github.com/apple/swift-syntax/blob/cecb0340a95f00f8f2725c912b494266ed7b4a80/Sources/SwiftBasicFormat/BasicFormat.swift#L177

**Comment by gibachan** [comment]
Thank you for your feedback. I'll take a look into it!

File: Sources/SwiftBasicFormat/BasicFormat.swift [hunk 1]
```diff
@@ -237,6 +236,13 @@ open class BasicFormat: SyntaxRewriter {
       (_, nil),
       (nil, _):
       return false
+    case (_, .colon):
+      let isTernaryExpr = second?.parent?.is(UnresolvedTernaryExprSyntax.self) == true
+      || second?.parent?.is(TernaryExprSyntax.self) == true
+
+      if !isTernaryExpr {
+        return false
+      }
```
Comment (ahoppen) [inline Sources/SwiftBasicFormat/BasicFormat.swift:245]: You can write this slightly  shorter by using `keyPathInParent`, just like we do in the second `switch` statement below.

```suggestion
    case (_, .colon):
      if second?.keyPathInParent != \TernaryExprSyntax.colonMark
        && second?.keyPathInParent != \UnresolvedTernaryExprSyntax.colonMark
      {
        return false
      }
```

**Review by ahoppen — COMMENTED** [review]
Thanks. This looks great! I’ve got one slight comment on how to improve implementation, otherwise this is ready to be merged. 👍🏽

File: Sources/SwiftBasicFormat/BasicFormat.swift (hunk 1 above)
Comment (gibachan) [reply Sources/SwiftBasicFormat/BasicFormat.swift]: Great! Thank you 👍

**Review by ahoppen — APPROVED** [review]
Thank you for your contribution, @gibachan. This looks good to me. Once CI passes, I’ll merge the PR.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**